### PR TITLE
MAINT: Remove unused compat functions

### PR DESCRIPTION
**Pull request by jbrockmendel**
Fully lint formula/ and compat/

### CODE REVIEW

**Comment by jbrockmendel** [comment]
@josef-pkt green and no merge conflict

**Comment by josef-pkt** [comment]
looks too much, and too many details that I don't like in quick skimming.
postponed

e.g. bytes versus string needs a more careful review and needs some modernizing.
This will be much easier when we drop python 2.7
e.g. I guess adding all the `b` for bytes might become obsolete when we use python 3 strings instead.

**Comment by jbrockmendel** [comment]
> looks too much, and too many details that I don't like in quick skimming.

Happy to break into smaller pieces.  Will wait for appveyor queue to die down.

> e.g. I guess adding all the b for bytes might become obsolete when we use python 3 strings instead.

I'll need to double-check, but I think a bunch of this is because of recarrays.

> This will be much easier when we drop python 2.7

Has this been decided upon?  numpy and pandas have made their drop dates explicit.  statsmodels tends to be slower in bumping version requirements.

**Comment by josef-pkt** [comment]
> e.g. I guess adding all the b for bytes might become obsolete when we use python 3 strings instead.
 > I'll need to double-check, but I think a bunch of this is because of recarrays.

It's not clear how we want to refactor this. eg. currently we use np.recfromtxt which required bytes when we wrote the test. AFAIK, but I didn't check, numpy got more string support.
Alternatively, we might want to switch to pandas eventually and drop recfromtxt.
In any case, I guess eventually we will want py 3 strings instead of bytes in these things.

("I guess", "I think", .... because I haven't looked at bytes versus strings in years)

File: statsmodels/compat/__init__.py [hunk 1]
```diff
@@ -1 +1,24 @@
-from .python import *
```
Comment (josef-pkt) [inline statsmodels/compat/__init__.py]: I find this replacement too painful to maintain, i.e. we have to change the same function name 3 times when there are changes.

Comment (jbrockmendel) [reply statsmodels/compat/__init__.py]: The `__all__` bit I get, but the `*` import is a pattern worth avoiding.

Comment (josef-pkt) [reply statsmodels/compat/__init__.py]: > but the * import is a pattern worth avoiding.

not in `__init__` or `api.py` when it's not important.

(In exclipse, I get some false red for imports from other packages, e.g. scipy or pytest, that doesn't recognize that a function is available there. I don't know whether that's related to `*` imports.)

Comment (jbrockmendel) [reply statsmodels/compat/__init__.py]: > not in __init__ or api.py when it's not important.

Its easy and it helps avoid mistakes.  e.g. are you sure the `compat` namespace is the same in py2/py3?  

Of course it's not a big deal.  All of these things are individually not-big-deals.  In aggregate they make for more readable, more maintainable code.

**Pull request closed without merging**